internal/tools/networkagent/main.go

```diff
@@ -0,0 +1,350 @@
+		}
+	}()
+
+	return &nodenetsvc.ConfigureNetworkingResponse{}, nil
+
+}
+
+func (s *service) teardownHelper(ctx context.Context, req *nodenetsvc.ConfigureNetworkingRequest, containerNamespaceID string) (*nodenetsvc.ConfigureNetworkingResponse, error) {
+	return s.teardownHNSHelper(ctx, req, containerNamespaceID)
+}
+
+func (s *service) teardownHNSHelper(ctx context.Context, req *nodenetsvc.ConfigureNetworkingRequest, containerNamespaceID string) (*nodenetsvc.ConfigureNetworkingResponse, error) {
+	eReq := &ncproxygrpc.GetEndpointsRequest{}
+	resp, err := s.client.GetEndpoints(ctx, eReq)
+	if err != nil {
+		return nil, err
+	}
+	for _, endpoint := range resp.Endpoints {
+		if endpoint.Namespace == containerNamespaceID {
+			nicID, ok := s.endpointToNicID[endpoint.Name]
+			if !ok {
+				log.G(ctx).WithField("name", endpoint.Name).Warn("endpoint was not assigned a NIC ID previously")
+				continue
+			}
+			// remove endpoints that are in the namespace as NICs
+			nsReq := &ncproxygrpc.DeleteNICRequest{
+				ContainerID:  req.ContainerID,
+				NicID:        nicID,
+				EndpointName: endpoint.Name,
+			}
+			if _, err := s.client.DeleteNIC(ctx, nsReq); err != nil {
+				log.G(ctx).WithField("name", endpoint.Name).Warn("failed to delete endpoint nic")
+			}
+			delete(s.endpointToNicID, endpoint.Name)
+		}
+	}
+	return &nodenetsvc.ConfigureNetworkingResponse{}, nil
+}
+
+func (s *service) ConfigureNetworking(ctx context.Context, req *nodenetsvc.ConfigureNetworkingRequest) (*nodenetsvc.ConfigureNetworkingResponse, error) {
+	log.G(ctx).WithField("req", req).Info("ConfigureNetworking request")
+
+	containerNamespaceID, ok := s.containerToNamespace[req.ContainerID]
+	if !ok {
+		return nil, fmt.Errorf("no namespace was previously created for containerID %s", req.ContainerID)
+	}
+
+	if req.RequestType == nodenetsvc.RequestType_Setup {
+		return s.addHelper(ctx, req, containerNamespaceID)
+	}
+	return s.teardownHelper(ctx, req, containerNamespaceID)
+}
+
+func (s *service) GetHostLocalIpAddress(ctx context.Context, req *nodenetsvc.GetHostLocalIpAddressRequest) (*nodenetsvc.GetHostLocalIpAddressResponse, error) {
+	return &nodenetsvc.GetHostLocalIpAddressResponse{IpAddr: ""}, nil
+}
+
+func (s *service) PingNodeNetworkService(ctx context.Context, req *nodenetsvc.PingNodeNetworkServiceRequest) (*nodenetsvc.PingNodeNetworkServiceResponse, error) {
+	return &nodenetsvc.PingNodeNetworkServiceResponse{}, nil
+}
+
+func main() {
+	var err error
+	ctx := context.Background()
+
+	flag.Parse()
+	conf, err := readConfig(*configPath)
+	if err != nil {
+		log.G(ctx).WithError(err).Fatalf("failed to read network agent's config file at %s", *configPath)
+		os.Exit(1)
+	}
+	log.G(ctx).WithFields(logrus.Fields{
+		"config path": *configPath,
+		"conf":        conf,
+	}).Info("network agent configuration")
+
+	sigChan := make(chan os.Signal, 1)
+	serveErr := make(chan error, 1)
+	defer close(serveErr)
+	signal.Notify(sigChan, syscall.SIGINT)
```

Comment:
```suggestion
	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
```

As I'd changed to this for the main ncproxy executable when I realized that they work on Windows. `os.Interrupt` will catch ctrl-c and ctrl-break, while sigterm will catch CTRL_CLOSE_EVENT, CTRL_LOGOFF_EVENT or CTRL_SHUTDOWN_EVENT. https://github.com/microsoft/hcsshim/blob/f20a064a5395e27dbd6634b367e94bd50bd8c2d1/cmd/ncproxy/run.go#L206